crates/iceberg/src/spec/view_metadata.rs

```diff
@@ -0,0 +1,682 @@
+    /// The view's base location; used to create metadata file locations
+    pub(crate) location: String,
+    /// ID of the current version of the view (version-id)
+    pub(crate) current_version_id: i64,
+    /// A list of known versions of the view
+    pub(crate) versions: HashMap<i64, ViewVersionRef>,
+    /// A list of version log entries with the timestamp and version-id for every
+    /// change to current-version-id
+    pub(crate) version_log: Vec<ViewVersionLog>,
+    /// A list of schemas, stored as objects with schema-id.
+    pub(crate) schemas: HashMap<i32, SchemaRef>,
+    /// A string to string map of view properties.
+    /// Properties are used for metadata such as comment and for settings that
+    /// affect view maintenance. This is not intended to be used for arbitrary metadata.
+    pub(crate) properties: HashMap<String, String>,
+}
+
+impl ViewMetadata {
+    /// Returns format version of this metadata.
+    #[inline]
+    pub fn format_version(&self) -> ViewFormatVersion {
+        self.format_version
+    }
+
+    /// Returns uuid of current view.
+    #[inline]
+    pub fn uuid(&self) -> Uuid {
+        self.view_uuid
+    }
+
+    /// Returns view location.
+    #[inline]
+    pub fn location(&self) -> &str {
+        self.location.as_str()
+    }
+
+    /// Returns the current version id.
+    #[inline]
+    pub fn current_version_id(&self) -> i64 {
+        self.current_version_id
+    }
+
+    /// Returns all view versions.
+    #[inline]
+    pub fn versions(&self) -> impl Iterator<Item = &ViewVersionRef> {
+        self.versions.values()
+    }
+
+    /// Lookup a view version by id.
+    #[inline]
+    pub fn version_by_id(&self, version_id: i64) -> Option<&ViewVersionRef> {
+        self.versions.get(&version_id)
+    }
+
+    /// Returns the current view version.
+    #[inline]
+    pub fn current_version(&self) -> &ViewVersionRef {
+        self.versions
+            .get(&self.current_version_id)
+            .expect("Current version id set, but not found in view versions")
+    }
+
+    /// Returns schemas
+    #[inline]
+    pub fn schemas_iter(&self) -> impl Iterator<Item = &SchemaRef> {
+        self.schemas.values()
+    }
+
+    /// Lookup schema by id.
+    #[inline]
+    pub fn schema_by_id(&self, schema_id: SchemaId) -> Option<&SchemaRef> {
+        self.schemas.get(&schema_id)
+    }
+
+    /// Get current schema
+    #[inline]
+    pub fn current_schema(&self) -> &SchemaRef {
+        let schema_id = self.current_version().schema_id();
+        self.schema_by_id(schema_id)
+            .expect("Current schema id set, but not found in view metadata")
+    }
+
+    /// Returns properties of the view.
+    #[inline]
+    pub fn properties(&self) -> &HashMap<String, String> {
+        &self.properties
+    }
+
+    /// Append view version to view
+    pub fn append_version(&mut self, view_version: ViewVersion) {
+        self.current_version_id = view_version.version_id();
```

Comment:
I feel that things like view version id re-assignment and view version de-duplication are missing in the builder.
The same applies for schema id re-assignment / schema de-duplication.

There are a bunch of tests in the java ref impl in `TestViewMetadata` that make sure these things behave correctly

Reply:
Yes, you are right. Bad idea to implement this half-done.
I am removing this function and handle it properly later in the full implementation of the builders.